client/my-sites/post-share/index.jsx

```diff
@@ -104,74 +154,206 @@ class PostShare extends Component {
+	}
+
+	renderRequestSharingNotice() {
+		const {
+			failure,
+			requesting,
+			success,
+			translate,
+		} = this.props;
+
+		if ( requesting ) {
+			return (
+				<Notice status="is-warning" showDismiss={ false }>
+						{ translate( 'Sharing…' ) }
+				</Notice>
+			);
+		}
+
+		if ( success ) {
+			return (
+				<Notice status="is-success" onDismissClick={ this.dismiss }>
+					{ translate( 'Post shared. Please check your social media accounts.' ) }
+				</Notice>
+			);
+		}
+
+		if ( failure ) {
+			return (
+				<Notice status="is-error" onDismissClick={ this.dismiss }>
+					{ translate( 'Something went wrong. Please don\'t be mad.' ) }
+				</Notice>
+			);
+		}
 	}
 
 	render() {
 		if ( ! this.props.isPublicizeEnabled ) {
 			return null;
 		}
 
-		if ( ! this.props.planHasRepublicizeFeature ) {
-			return ( <div className="post-share">
-				<Banner
-					feature="republicize"
-					title={ this.props.translate( 'Unlock the ability to re-share posts to social media' ) }
-					callToAction={ this.props.translate( 'Upgrade to Premium' ) }
-					description={ this.props.translate( 'Get unlimited premium themes, video uploads, monetize your site and more.' ) }
-				/>
-			</div> );
+		if ( ! this.props.hasRepublicizeFeature ) {
+			return this.renderUpgradeToGetPublicizeNudge();
 		}
 
-		const classes = classNames( 'post-share__wrapper', {
-			'has-connections': this.hasConnections()
-		} );
+		const {
+			connections,
+			postId,
+			siteId,
+			siteSlug,
+			translate,
+		} = this.props;
 
-		return (
-			<div className="post-share">
-				{ this.props.requesting &&
-					<Notice status="is-warning" showDismiss={ false }>
-							{ this.props.translate( 'Sharing…' ) }
-					</Notice>
-				}
+		if ( ! siteId || ! postId ) {
+			return null;
+		}
 
-				{ this.props.success &&
-					<Notice status="is-success" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Post shared. Please check your social media accounts.' ) }
-					</Notice>
-				}
+		const classes = classNames(
+			'post-share__wrapper',
+			{ 'has-connections': this.hasConnections() },
+			{ 'has-republicize-feature': this.props.hasRepublicizeFeature },
+			{ 'has-republicize-scheduling-feature': this.props.hasRepublicizeSchedulingFeature },
+			{ 'has-connections': this.hasConnections() },
+		);
 
-				{ this.props.failure &&
-					<Notice status="is-error" onDismissClick={ this.dismiss }>
-						{ this.props.translate( 'Something went wrong. Please don\'t be mad.' ) }
-						</Notice>
-				}
+		return (
+			<div className="post-share">
+				{ this.renderRequestSharingNotice() }
 
 				<div className={ classes }>
-					{ this.props.siteId && <QueryPostTypes siteId={ this.props.siteId } /> }
+					<QueryPublicizeConnections siteId={ siteId } />
+					<QueryPostTypes siteId={ siteId } />
+
 					<div className="post-share__head">
 						<h4 className="post-share__title">
-							{ this.props.translate( 'Share post' ) }
+							{ translate( 'Share this post' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **7** times:
`translate( 'Share this post!' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).